Makefile

```diff
@@ -31,10 +20,24 @@ SERVICES := omniscidb postgres mysql clickhouse impala kudu-master kudu-tserver
 # the variable contains backends for which test datasets can be automatically loaded
 LOADS := sqlite parquet postgres clickhouse omniscidb mysql impala
 
+# which backends we want to install dependencies for
+INSTALL_DEPS_FOR := bigquery clickhouse impala mysql omniscidb parquet postgres
+
```

Comment:
The CI is split in two parts:

1. GitHub Actions
2. Azure pipeline

GitHub actions is fine--for each backend it tests, it **only** installs requirements from the dependency files it needs.

The Azure pipeline however (which currently tests Impala and Clickhouse and builds the docs) installs dependencies for **all backends**. This now causes the CI to fail, because now, the dependencies for PySpark and Spark are not compatible with the dependencies for other backends

This change here allows us to limit which backends we are installing dependencies for in the `ibis` Docker image, which the Azure pipeline uses. (Setting the default ^ to be to install dependencies for all backends except for the PySpark and Spark backends.)

Reply:
The reason the dependencies for all backends are being installed is for the docs. To obtain the docstrings, the code need to be loaded, and to load the code/docstrings for all backends we need all the dependencies installed.

I didn't check your changes in detail, but you may want to check that the spark documentation continues to be rendered fine after this change.

Reply:
Yes that makes sense--I did need to make a change to the docs build to get it to pass:
1. Created a [`ci/deps/docs.yml`](https://github.com/ibis-project/ibis/blob/758f09bee2a469678e818df85a8858103c2c029d/ci/deps/docs.yml) file containing the dependencies for all backends in one file, tweaking package versions so that there are no conflicts
2. In the docs build, [use this new `docs.yml` when building the `ibis` image](https://github.com/ibis-project/ibis/pull/2348/files#diff-903fe327761cfb535a196f2c04326e43R110)

It's not amazing that every backend-specific dependency would need to be updated/maintained in two places now. An alternative would be to make `docs.yml` an "override" requirements file, containing only adjustments that need to be made to packages to prevent conflicts, but I felt like that would have made the CI more complex. This seems like the simplest way